foundation/storage/README.md

```diff
@@ -7,66 +7,84 @@
 
 # Ping Storage SDK
 
-The Ping Storage SDK provides a flexible storage interface and a set of common storage solutions for
-the Ping SDKs.
+The Ping Storage SDK provides a flexible storage interface and a set of common storage solutions for the Ping SDKs.
+
+---
+
+## Table of Contents
+
+- [Integrating the SDK into your project](#integrating-the-sdk-into-your-project)
+- [Storage Interface](#storage-interface)
+- [Available Storage Solutions](#available-storage-solutions)
+- [How to Use the SDK](#how-to-use-the-sdk)
+    - [Creating and Using a Storage Instance](#creating-and-using-a-storage-instance)
+    - [DataStoreStorage](#datastorestorage)
+    - [EncryptedDataStoreStorage](#encrypteddatastorestorage)
+    - [Enabling Cache for the Storage](#enabling-cache-for-the-storage)
+    - [Cache Strategy Table](#cache-strategy-table)
+- [Creating a Custom Storage](#creating-a-custom-storage)
+
+---
 
 ## Integrating the SDK into your project
 
-To add the Ping Storage SDK as a dependency to your project, include the following in
-your `build.gradle` file:
+To add the Ping Storage SDK as a dependency to your project, include the following in your `build.gradle` file:
 
 ```kotlin
 dependencies {
     implementation(project(":foundation:storage"))
```

Comment:
Customers would rather use the maven package... i.e:
`implementation("com.pingidentity.sdks:storage:<version>")`